### ISSUE TITLE
Should check for null pointers where Count > 0

### ISSUE THREAD

**Issue by jarikomppa** [KhronosGroup/Vulkan-ValidationLayers#699]
Layers should check for null pointers in structures where count of zero ignores the pointer.

For example VkSubmitInfo has waitSemaphoreCount, commandBufferCount and signalSemaphoreCount. If the counts are by accident set to non-zero while their corresponding pointers are NULL, the layers will crash.

**Comment by tobine** [comment]
Are you running with parameter validation on the top of your layer stack?
Do you see a validation error from that layer prior to the crash?
Parameter validation should flag an error on the invalid semaphore and/or command buffer handles prior to any crash.
Once you hit a layer error, behavior is undefined and crashes may occur. In general lower level layers (such as core_validation) don't have safeguards against invalid parameters or object handles that are caught by parameter_validation and object_tracker layers.

**Comment by jarikomppa** [comment]
I'll have to double check this. I stumbled on this problem while developing a test for the vk-gl-cts. I think I have all related layers enabled, but still got a crash instead of a layer error.

**Comment by jarikomppa** [comment]
Either the layers are not enabled right for me, or I'm running too old layers then. Sorry about the false alarm.

**Comment by mark-lunarg** [comment]
If you have any questions or find out that we don't know what we're talking about, send us a repro case and we'll address it straightaway -- thanks!

**Comment by jarikomppa** [comment]
FWIW, I still can't seem to reproduce the null pointer check. If I set waitsemaphores or signalsemaphores to null pointer, I get a crash in VkLauer_thread_safety.dll; if I set commandbuffers to null pointer, I get a crash in VkLayer_object_lifetimes.dll. Using VulkanSDK 1.1.97.0, according to what registry says.

If I skip setting the VkSubmitInfo.sType, I do get error from the layer, so it's there.
